tests/settings/test_util.py

```diff
@@ -0,0 +1,69 @@
+import json
+
+import pytest
+
+import py42.settings as settings
+from py42.response import Py42Response
+from py42.services.util import get_all_pages
+
+
+@pytest.fixture
+def empty_response(mocker):
+    response = mocker.MagicMock(spec=Py42Response)
+    response.status_code = 200
+    response.text = '{"items": []}'
+    response.__getitem__ = lambda _, key: json.loads(response.text)[key]
+    return response
+
+
+@pytest.fixture
+def get_three_three_item_pages(mocker, py42_response, empty_response):
+    py42_response.text = '{"items": [1, 2, 3]}'
+    mock = mocker.MagicMock()
+    mock.side_effect = [py42_response, py42_response, py42_response, empty_response]
+    return mock
+
+
+def verify_calls(func, page_size):
+    assert func.call_args_list[0][1]["page_num"] == 1
+    assert func.call_args_list[1][1]["page_num"] == 2
+    assert func.call_args_list[2][1]["page_num"] == 3
+    assert func.call_args_list[3][1]["page_num"] == 4
+    assert func.call_args_list[0][1]["page_size"] == page_size
+    assert func.call_args_list[1][1]["page_size"] == page_size
+    assert func.call_args_list[2][1]["page_size"] == page_size
+    assert func.call_args_list[3][1]["page_size"] == page_size
+    assert func.call_count == 4
+
+
+def test_get_all_pages_with_no_input_page_size_calls_callback_with_expected_values(
+    get_three_three_item_pages,
+):
+    settings.items_per_page = 3
```

Comment:
This is a good candidate for having a test class that implements `__enter__` and `__exit__` and resets the page .
Doing that ensures that resetting the items per page happens every time,

```
class TestPageSizer:
    def __init__(self, test_count, reset_count=500):
        self.test_count = test_count
        self.reset_count = reset_count

    def __enter__(self, *args, **kwargs):
        settings.items_per_page = self.test_count

    def __exit__(self, *args, **kwargs):
        settings.items_per_page = self.reset_count
```

```
with TestPageSizer(3, 500):
   <test>
```

Reply:
Seeing resets in tests always make think of this

Reply:
totally agree, these tests are definitely a victim of copy/pasting from other places that did not do this

Reply:
Ah that is good to know, I will remember that